README.md

```diff
@@ -98,5 +105,67 @@ Insert the `Save Build Scan` step after each Maven execution step in the Github
       - name: Build with Maven
         run: mvn clean package
       - name: Save Build Scan
-        uses: gradle/github-actions/maven/build-scan-save@v1.0
-[...]```
+        uses: gradle/github-actions/maven/build-scan/save@v1.0
+[...]
+```
+
+## maven/build-scan/publish
+
+A composite action to publish all Maven Build Scans® saved as workflow artifacts when validating a pull-request (by the `maven/build-scan/save` action). 
+
+Use this action in a separate workflow with a `workflow_run` event trigger, that will run after an existing pull-request workflow has completed. The action will download any saved Build Scan® and publish it to Gradle Enterprise.
+This event allows access to the repository secrets (_Gradle Enterprise Access Key_) which is required to publish a Build Scan® to Gradle Enterprise when authentication is enabled.
+
+The Build Scan® publication requires the Gradle Terms of Service to be accepted, this can be achieved by adding a workflow using the `terms-of-service-acceptance/run` action.
+The `terms-of-service-acceptance/verify` action is used to ensure this workflow passed successfully. 
+
+`dawidd6/action-download-artifact` action is used to download Artifacts uploaded by a different workflow.
+
+**Dependencies**:
+
+- [dawidd6/action-download-artifact](https://github.com/marketplace/actions/download-workflow-artifact)
+- [terms-of-service-acceptance/verify](./terms-of-service-acceptance/verify/action.yml)
```

Comment:
This isn't a dependency of the `maven/build-scan/publish` action. These 2 actions work together to provide the full "publish workflows on PRs" functionality.